warehouse/query-core/src/main/java/datawave/query/jexl/visitors/RootNegationCheckVisitor.java

```diff
@@ -1,33 +1,61 @@
 package datawave.query.jexl.visitors;
 
+import datawave.query.jexl.nodes.QueryPropertyMarker;
 import org.apache.commons.jexl2.parser.ASTNENode;
 import org.apache.commons.jexl2.parser.ASTNRNode;
 import org.apache.commons.jexl2.parser.ASTNotNode;
 import org.apache.commons.jexl2.parser.JexlNode;
 
 /**
- * Determine if a negation occurs at the root of the AST
+ * Determine if all query terms are negated
  */
 public class RootNegationCheckVisitor {
     
+    /**
+     * Since negations may be pushed down instead of simply looking for a top level negation, look for a leaf node without a negation.
+     * 
+     * @param script
+     * @return true if there is no path to a leaf node without passing through a negation, false otherwise
+     */
     public static Boolean hasTopLevelNegation(JexlNode script) {
         RootNegationCheckVisitor visitor = new RootNegationCheckVisitor();
-        return visitor.isNegation(script);
+        
+        // ensure all negations are pushed down
+        JexlNode pushedDownTree = PushdownNegationVisitor.pushdownNegations(script);
+        return !visitor.hasLeafWithoutNegation(pushedDownTree);
     }
     
-    private boolean isNegation(JexlNode node) {
-        boolean hasNegation = false;
-        for (int i = 0; i < node.jjtGetNumChildren() && !hasNegation; i++) {
+    /**
+     * For each child skip over any property markers and look for negations and leafs. If neither is found examine that child until either a negation or leaf is
+     * found
+     * 
+     * @param node
+     * @return true if a leaf node was found, false otherwise
+     */
+    private boolean hasLeafWithoutNegation(JexlNode node) {
+        boolean foundLeaf = false;
+        for (int i = 0; i < node.jjtGetNumChildren() && !foundLeaf; i++) {
             JexlNode child = node.jjtGetChild(i);
             Class<?> childClass = child.getClass();
             
+            // skip over any query property markers/assignments
+            JexlNode sourceNode = QueryPropertyMarker.getQueryPropertySource(child, null);
+            if (sourceNode != null) {
+                child = sourceNode;
+            }
+            
             if (ASTNENode.class.equals(childClass) || ASTNotNode.class.equals(childClass) || ASTNRNode.class.equals(childClass)) {
```

Comment:
I know this wasn't your code, but .equals vs instanceof?  just a nit.

Reply:
I prefer instanceof as well.

Reply:
Are we pushing this off to a separate issue?

Reply:
Or do we just not care that much?  that's ok too i think.

Reply:
I think there is enough in this PR. I'd rather not make any out of scope changes.